packages/patrol_cli/test/ios/ios_test_backend_test.dart

```diff
@@ -1,40 +1,177 @@
+      const buildMode = BuildMode.debug;
+      const flavor = 'dev';
+
+      expect(buildMode.createConfiguration(flavor), 'Debug-dev');
+      expect(buildMode.createScheme(flavor), 'dev');
+    });
+
+    test('infers build options in release mode when flavor is null', () {
+      const buildMode = BuildMode.release;
+      String? flavor;
+
+      expect(buildMode.createConfiguration(flavor), 'Release');
+      expect(buildMode.createScheme(flavor), 'Runner');
+    });
+
+    test('infers build options in release mode when flavor is not null', () {
+      const buildMode = BuildMode.release;
+      const flavor = 'prod';
+
+      expect(buildMode.createConfiguration(flavor), 'Release-prod');
+      expect(buildMode.createScheme(flavor), 'prod');
+    });
+  });
+
   group('IOSTestBackend', () {
-    group('BuildMode', () {
-      test('infers build options in debug mode when flavor is null', () {
-        const buildMode = BuildMode.debug;
-        String? flavor;
-
-        expect(buildMode.createConfiguration(flavor), 'Debug');
-        expect(buildMode.createScheme(flavor), 'Runner');
-      });
-
-      test('infers build options in debug mode when flavor is not null', () {
-        const buildMode = BuildMode.debug;
-        const flavor = 'dev';
-
-        expect(buildMode.createConfiguration(flavor), 'Debug-dev');
-        expect(buildMode.createScheme(flavor), 'dev');
-      });
-
-      test('infers build options in release mode when flavor is null', () {
-        const buildMode = BuildMode.release;
-        String? flavor;
-
-        expect(buildMode.createConfiguration(flavor), 'Release');
-        expect(buildMode.createScheme(flavor), 'Runner');
-      });
-
-      test('infers build options in release mode when flavor is not null', () {
-        const buildMode = BuildMode.release;
-        const flavor = 'prod';
-
-        expect(buildMode.createConfiguration(flavor), 'Release-prod');
-        expect(buildMode.createScheme(flavor), 'prod');
-      });
+    late IOSTestBackend iosTestBackend;
+    late FileSystem fs;
+
+    setUp(() {
+      fs = MemoryFileSystem.test();
+      fs.directory('example_app').createSync();
+      fs.currentDirectory = 'example_app';
+
+      iosTestBackend = IOSTestBackend(
+        processManager: FakeProcessManager(),
+        fs: fs,
+        iosDeploy: FakeIOSDeploy(),
+        parentDisposeScope: DisposeScope(),
+        logger: FakeLogger(),
+      );
+    });
+
+    group('xcTestRunPath', () {
+      @isTest
+      void testXcTestRunPath(
+        String description, {
+        String scheme = 'Runner',
+        bool simulator = false,
+        String? arch,
+      }) {
+        test(description, () async {
+          final targetPlatform = simulator ? 'iphonesimulator' : 'iphoneos';
+          var targetArch = arch;
+          if (targetArch != null) {
+            targetArch = '-$targetArch';
+          }
+
+          final name = '${scheme}_${targetPlatform}16.2$targetArch.xctestrun';
```

Comment:
I don't like having this in test, it's essentially duplicated code from `IOSTestBackend.xcTestRunPath`.